src/index.ts

```diff
@@ -12,39 +17,50 @@ declare global {
 }
 
 // List of required features browser features
-const requiredFeatures = ["Worker", "Promise", "fetch"];
+const requiredFeatures = ["Promise", "fetch"];
 // Test whether browser has required feature support
-const hasFeatureSupport = hasProperties(window, requiredFeatures);
+const hasFeatureSupport =
+  hasProperties(window, requiredFeatures) &&
+  "getEntriesByType" in performance &&
+  typeof performance.getEntriesByType === "function";
 
 // Define global state
 const state: Fastly = {
   client: {
-    hasFeatureSupport
+    hasFeatureSupport,
   },
   // The are certain scenarios (such as the demo website) in which the global
   // FASTLY and the results array already exists.
-  results: (window.FASTLY && window.FASTLY.results) || []
+  results: (window.FASTLY && window.FASTLY.results) || [],
 };
 
-// Init initializes the library when the browser is ready
-export function init(): void {
-  loadWhenDocumentReady((): void => {
-    const params = getScriptParameters(SCRIPT_SRC_REGEXP);
-    worker
-      .init(params)
-      .then((results: Beacon[]): void => {
-        // We use apply here as we don't want to directly assign i.e.
-        // replace the existing results array.
-        Array.prototype.push.apply(state.results, results);
-      })
-      .catch((): void => {});
-  });
+function paramsToConfigUrl({ host, k: token }: QueryParameters): string {
+  const configHost =
+    host === "https://www.fastly-insights.com"
+      ? "https://fastly-insights.com"
+      : host;
+  return `${configHost}${CONFIG_PATH}${token}`;
 }
 
 // If browser has feature support,
 // initialize the library
 if (hasFeatureSupport) {
-  init();
+  const params = getScriptParameters(SCRIPT_SRC_REGEXP);
+
+  const settingsBuilder = new ClientSettingsBuilder();
+  const settings: Settings = {
+    config_url: paramsToConfigUrl(params),
+    report_errors: true,
+    token: params.token,
+  };
+
+  settingsBuilder.addProvider(new FastlyProvider(settings));
+
+  init(settingsBuilder.toSettings()).then((result) => {
+    // We use apply here as we don't want to directly assign i.e.
+    // replace the existing results array.
+    Array.prototype.push.apply(state.results, result.testResults);
```

Comment:
Oh, I haven't come across `Array.prototype.push.apply` in a while :) Are we using this to appease the linter? If so, I'd like to offer a nit-picky suggestion to save some code comments and/or improve readability:
```suggestion
    state.results.push(...result.testResults);
```
The targets in `.browserlistrc` mean that this gets transpiled to `state.results.apply(state.results, result.testResults)` anyway, which is equivalent to `Array.prototype.push.apply(state.results, result.testResults);`. Untranspiled (ESM version?), `arr.push(...extraElements)` [is marginally faster than](https://www.measurethat.net/Benchmarks/Show/2279/0/arrayprototypeconcat-vs-spread-operator-vs-apply) `Apply.prototype.push.apply(arr,extraElements)`.

Reply:
Apologies, the comment above this line should be more descriptive as to why we **HAVE** to use `Array.prototype.push.apply()` in this instance. 

For the https://insights.fastlylabs.com website we have a demo application which loads insights.js via a script tag and registers a `new Proxy()` on the window global `FASTLY.results`, so that we can spy on the results without manipulating the third-party script. 

TL;DR you use `Array.prototype.push.apply()` when you're interested in using `.push()` with an object thats isn't really an Array.